Add maybe_unused specifier for unused loop var

mrodozov opened the pull request:
this test part was introducing unused variable https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_aarch64_gcc700/CMSSW_10_2_X_2018-06-03-2300/CondCore/SiStripPlugins

Commit:
Add maybe_unused specifier for unused loop var

mrodozov commented:
please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23442/4977

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @mrodozov (Mircho Rodozov) for master.

It involves the following packages:

CondCore/SiStripPlugins


@ggovi, @cmsbuild can you please review it and eventually sign? Thanks.
@mmusich, @VinInn this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
+1
Tested at: 9607cfb43b49326d6e176cd28f11e3de87a1fd22
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23442/28412/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23442/28412/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902471
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2902279
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

ggovi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented:
@mrodozov @smuzaffar @Dr15Jones what is the expected behaviour here? Looking at the compilation  result, I see that the warning is still reported

2300/src/CondCore/SiStripPlugins/plugins/SiStripDetVOff_PayloadInspector.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/CondCore/SiStripPlugins/plugins/SiStripLatency_PayloadInspector.cc: In member function 'virtual bool {anonymous}::SiStripLatencyTest::fill(const std::vector<std::tuple<long long unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&)':
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-06-03-2300/src/CondCore/SiStripPlugins/plugins/SiStripLatency_PayloadInspector.cc:62:39: warning: unused variable 'l' [-Wunused-variable]
    for ([[maybe_unused]] const auto & l : lat) {
                                       ^

mmusich commented:
@fabiocos, I believe this is a leftover by @jpriscia PR https://github.com/cms-sw/cmssw/pull/23316.
The `SiStripLatencyTest` at the moment is useless. I suggest to either remove it or wait for @jpriscia to rewrite it in a meaningful way.

smuzaffar commented:
hold
this change only work for gcc 7 and above (c++17). I will let @jpriscia  to decide what to do here.

cmsbuild commented:
Pull request has been put on hold by  @smuzaffar
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

Commit:
Update SiStripLatency_PayloadInspector.cc

I'm proposing this as temp solution until we decide should or shouldn't remove the entire block. I didn't remove it because I didn't want to 'guess' what is appropriate to add/remove.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23442/5069

cmsbuild commented:
Pull request #23442 was updated. @ggovi, @cmsbuild can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 1f79b25254c3e14a8fabce84e69dc43346877bfd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23442/28515/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23442/28515/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902639
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2902447
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

mmusich commented:
@mrodozov @smuzaffar I fixed the issue together with another one in #23514.
As this PR is superseded please close it. Thanks

smuzaffar commented:
Thansk @mmusich.
Closing this Pr in favour of #23514
